swift_browser_ui/upload/cryptupload.py

```diff
@@ -85,6 +84,43 @@ def check_header(
     ) -> bool:
         return self.header_uploaded
 
+    async def a_create_container(self) -> None:
+        """Create the container required by the upload."""
+        async with self.client.head(
+            common.generate_download_url(self.host, self.container),
+            headers={"Content-Length": "0", "X-Auth-Token": self.token},
+            ssl=ssl_context,
+        ) as resp_get:
+            if resp_get.status != 204:
+                async with self.client.put(
+                    common.generate_download_url(self.host, self.container),
+                    headers={"Content-Length": "0", "X-Auth-Token": self.token},
+                    ssl=ssl_context,
+                ) as resp_put:
+                    if resp_put.status not in {201, 202}:
+                        raise aiohttp.web.HTTPForbidden(
+                            reason="Upload container creation failed."
+                        )
+
+        async with self.client.head(
```

Comment:
These 2 blocks look almost the same. Could they be made into one, for e.g. with 
`for container in {self.container, f"{self.container}_segments"}:`

Reply:
True, that'd work – I'll add the change 👍🏼